### PR TITLE
Added CouchDB storage

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
Closes https://github.com/doctrine/KeyValueStore/issues/40

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     },
     "require-dev": {
         "phpunit/phpunit": "^4.8",
-        "riak/riak-client": "dev-master"
+        "riak/riak-client": "dev-master",
+        "doctrine/couchdb": "^1.0.0-beta2"
```
Comment (Ocramius) [inline composer.json]: Is there a more stable version now?

File: lib/Doctrine/KeyValueStore/Storage/CouchDbStorage.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+/**
+ * Key-Value-Storage using a Doctrine CouchDB Client library as backend.
+ *
+ * CouchDB requires scalar identifiers, so the `flattenKey` method is used
+ * to flatten passed keys.
+ *
+ * @author Emanuele Minotto <minottoemanuele@gmail.com>
+ */
+class CouchDbStorage implements Storage
```
Comment (Ocramius) [inline lib/Doctrine/KeyValueStore/Storage/CouchDbStorage.php]: `final`

File: lib/Doctrine/KeyValueStore/Storage/CouchDbStorage.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+     * @param string       $storageName
+     * @param array|string $key
+     * 
+     * @return string
+     */
+    private function flattenKey($storageName, $key)
+    {
+        $finalKey = $storageName . '-';
+
+        if (is_scalar($key)) {
```
Comment (Ocramius) [inline lib/Doctrine/KeyValueStore/Storage/CouchDbStorage.php]: Should be `is_string`

File: lib/Doctrine/KeyValueStore/Storage/CouchDbStorage.php [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            return $finalKey . $key;
+        }
+
+        foreach ($key as $property => $value) {
```
Comment (Ocramius) [inline lib/Doctrine/KeyValueStore/Storage/CouchDbStorage.php]: What if the key is not an `array` at this point? Should have a failure...

File: tests/Doctrine/Tests/KeyValueStore/Storage/CouchDbStorageTest.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Doctrine\Tests\KeyValueStore\Storage;
+
+use Doctrine\KeyValueStore\Storage\CouchDbStorage;
+
+/**
+ * CouchDb storage testcase
+ *
+ * @author Emanuele Minotto <minottoemanuele@gmail.com
```
Comment (Ocramius) [inline tests/Doctrine/Tests/KeyValueStore/Storage/CouchDbStorageTest.php]: Add `@covers` here, to prevent coverage leakage

**Comment by beberlei** [comment]
CouchDB Supports range queries, so you could add support for that.

File: composer.json (hunk 1 above)
Comment (EmanueleMinotto) [inline composer.json]: Currently there's the beta 4 but the stability level is still that.

**Comment by Ocramius** [comment]
:+1:

**Pull request merged**